worker/changestream/manifold.go

```diff
@@ -95,20 +108,32 @@ func Manifold(config ManifoldConfig) dependency.Manifold {
 				return nil, errors.Trace(err)
 			}
 
+			// Register the metrics collector against the prometheus register.
+			metricsCollector := config.NewMetricsCollector()
+			if err := config.PrometheusRegisterer.Register(metricsCollector); err != nil {
+				return nil, errors.Trace(err)
+			}
+
 			cfg := WorkerConfig{
 				AgentTag:          agentConfig.Tag().Id(),
 				DBGetter:          dbGetter,
 				FileNotifyWatcher: fileNotifyWatcher,
 				Clock:             config.Clock,
 				Logger:            config.Logger,
+				Metrics:           metricsCollector,
```

Comment:
This should be `metricsCollector.ForModel()` then you can get rid of all the metrics at the top level and just pass the model one around.